### PR TITLE
CED-1815  Fix accessibility issue in v3 EsRadioButton

### PR DESCRIPTION
**Pull request by tomleo**
### 🔗 Linked issue

- https://energysage.atlassian.net/browse/CED-1815

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Renamed `EsRadioButton` to `EsRadioInput` as there's no buttons involved with this component. The odd naming is an artifact of the radio component it's wrapping [PrimeVue RadioButton](https://v3.primevue.org/radiobutton/)
- Added a `EsRadioInputGroup` component to match how we've been using `b-form-radio-group` [see JIRA for comtext](https://energysage.atlassian.net/browse/CED-1815?focusedCommentId=46633)
- Wrapped all `EsRadioInput` examples in `EsRadioInputGroup` components to ensure fieldset & legend are present (for a11y reasons)

### 🥼 Testing

- Manual

#### 🧐 Feedback Requested / Focus Areas

- When going through implementation usage across downstream repos, we make heavy use of `b-form-group` as well. I started to implement a similar component, but that seemed out-of-scope for this. Something we may want to implement down the line though.
- I removed the area-labledby prop present in the bootstrap-vue's radio-input-group, based on a11y documentation https://www.w3.org/WAI/tutorials/forms/grouping/#radio-buttons I think the fieldset should be the proper way to handle this.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

**Review by ericdouglaspratt — COMMENTED** [review]
Radio button is a pretty standard name across the entire industry ([Wikipedia](https://en.wikipedia.org/wiki/Radio_button)) that has a specific etymology. It's also called that in the [ESDS 2.0 Figma](https://www.figma.com/design/G3Zq89yg3vek9b4JrnlSYn/ESDS-2.0?node-id=17102-2188&node-type=FRAME&t=jE95379y6wZilGHU-0).

I'd recommend keeping the name, at least on the docs site and in the UI.

**Comment by tomleo** [comment]
> Radio button is a pretty standard name across the entire industry ([Wikipedia](https://en.wikipedia.org/wiki/Radio_button)) that has a specific etymology. It's also called that in the [ESDS 2.0 Figma](https://www.figma.com/design/G3Zq89yg3vek9b4JrnlSYn/ESDS-2.0?node-id=17102-2188&node-type=FRAME&t=jE95379y6wZilGHU-0).
> 
> I'd recommend keeping the name, at least on the docs site and in the UI.

Oh weird, bootstrap seems to follow the actual markup `input type="radio"` with their naming convention of `b-form-radio`, but aligning with Figma is probably most important. I'll revert back.

File: es-ds-docs/pages/molecules/radio-button.vue [hunk 1]
```diff
@@ -51,51 +79,86 @@ const propTableRows = [
...
+                    v-model="selectedFruit"
+                    :display-name="fruit.name"
+                    group-name="disabled"
+                    :value="fruit.key"
+                    disabled />
+            </es-radio-button-group>
+        </div>
+
+        <div class="my-500">
+            <h2>Using Options Prop</h2>
```
Comment (hroth1994) [inline es-ds-docs/pages/molecules/radio-button.vue:128]: Sentence case - Using options prop

File: es-ds-docs/pages/molecules/radio-button.vue [hunk 2]
```diff
@@ -51,51 +79,86 @@ const propTableRows = [
...
+
+        <div class="mb-500">
+            <h2>EsRadioButtonGroup props</h2>
+            <ds-prop-table :rows="radioButtonGroupPropTableRows" />
         </div>
 
         <ds-doc-source
-            :comp-code="compCode"
-            comp-source="es-ds-components/src/lib-components/es-radio-button.vue"
+            comp-title="Radio Button Component"
```
Comment (hroth1994) [inline es-ds-docs/pages/molecules/radio-button.vue:150]: Sentence case - Radio button component

File: es-ds-docs/pages/molecules/radio-button.vue [hunk 3]
```diff
@@ -51,51 +79,86 @@ const propTableRows = [
...
+            :comp-code="radioButtonComponentCode"
+            comp-source="es-ds-components/src/lib-components/es-radio-button.vue" />
+
+        <ds-doc-source
+            comp-title="Radio Button Group Component"
```
Comment (hroth1994) [inline es-ds-docs/pages/molecules/radio-button.vue:155]: Sentence case - Radio button group component

File: es-ds-docs/pages/molecules/radio-button.vue [hunk 4]
```diff
@@ -51,51 +79,86 @@ const propTableRows = [
...
+            :comp-code="radioButtonGroupComponentCode"
+            comp-source="es-ds-components/src/lib-components/es-radio-button-group.vue" />
+
+        <ds-doc-source
+            doc-title="Radio Button & Group Documentation"
             :doc-code="docCode"
-            doc-source="es-ds-docs/pages/molecules/radio-button.vue" />
+            doc-source="es-ds-docs/pages/molecules/radio-input.vue" />
```
Comment (hroth1994) [inline es-ds-docs/pages/molecules/radio-button.vue:162]: This should be `es-ds-docs/pages/molecules/radio-button.vue`

File: es-ds-docs/pages/molecules/radio-button.vue [hunk 5]
```diff
@@ -51,51 +79,86 @@ const propTableRows = [
...
+            :comp-code="radioButtonComponentCode"
+            comp-source="es-ds-components/src/lib-components/es-radio-button.vue" />
+
+        <ds-doc-source
+            comp-title="Radio Button Group Component"
+            :comp-code="radioButtonGroupComponentCode"
+            comp-source="es-ds-components/src/lib-components/es-radio-button-group.vue" />
+
+        <ds-doc-source
+            doc-title="Radio Button & Group Documentation"
```
Comment (hroth1994) [inline es-ds-docs/pages/molecules/radio-button.vue:160]: Sentence case - or maybe Radio button & Radio group documentation?

**Review by hroth1994 — COMMENTED** [review]
A few comments on the documentation

File: es-ds-docs/pages/molecules/radio-button.vue [hunk 6]
```diff
@@ -35,11 +30,44 @@ const propTableRows = [
             'use the same v-model.',
     ],
 ];
+
+// Name, Type, Default, Description
+const radioButtonGroupPropTableRows = [
+    ['id', 'String', '', 'Required.'],
```
Comment (hroth1994) [inline es-ds-docs/pages/molecules/radio-button.vue:36]: Can you put n/a under Default value for required fields?

File: es-ds-docs/pages/molecules/radio-button.vue (hunk 4 above)
Comment (tomleo) [reply es-ds-docs/pages/molecules/radio-button.vue]: 😓 good catch!

**Review by hroth1994 — APPROVED** [review]
LGTM! 🌟

**Pull request merged**